Bluetooth: drivers: Sync bus types with BlueZ

jhedberg opened the pull request:
The authoritative source of these values is BlueZ:

https://git.kernel.org/pub/scm/bluetooth/bluez.git/tree/lib/hci.h#n38

Update our values with the above. The IPM definiton doesn't exist in BlueZ, but should be added there to make sure we don't get out of sync again.

Fixes #80984

jhedberg committed:
Bluetooth: drivers: Sync bus types with BlueZ

The authoritative source of these values is BlueZ:

https://git.kernel.org/pub/scm/bluetooth/bluez.git/tree/lib/hci.h#n38

Update our values with the above. The IPM definiton doesn't exist in
BlueZ, but should be added there to make sure we don't get out of sync
again.

Signed-off-by: Johan Hedberg <johan.hedberg@silabs.com>

jhedberg commented:
@Vudentz any chance you could help add the IPM definition to BlueZ? That way, if BlueZ extends this list, at least we don't use overlapping values. Likewise, if we ever need to extend it, we'll make sure to get the same addition to BlueZ as well.

alwa-nordic commented:
Should we add a code comment with a link to the authoritative source?

jhedberg commented:
> Should we add a code comment with a link to the authoritative source?

I'm just not sure how to make that reference so that it stays up to date, besides simply saying "These values come from Linux/BlueZ". What makes the situation more confusing is that BlueZ has multiple places defining them. There's e.g. the following which seems to be out of date with the latest VirtIO addition (@Vudenz you might want to fix this):
https://git.kernel.org/pub/scm/bluetooth/bluez.git/tree/src/shared/btsnoop.h#n51

The Linux kernel also has its own definitions, which could be seen as equally authoritative as the user space tree:
https://git.kernel.org/pub/scm/bluetooth/bluetooth-next.git/tree/include/net/bluetooth/hci.h#n60

carlescufi commented on include/zephyr/drivers/bluetooth.h:
```diff
@@ -69,7 +69,9 @@ enum bt_hci_bus {
-	BT_HCI_BUS_IPM           = 9,
+	BT_HCI_BUS_SMD           = 9,
+	BT_HCI_BUS_VIRTIO        = 10,
+	BT_HCI_BUS_IPM           = 11,
```
I would recommend instead using `IPC` instead of `IPM`, since `IPM` is a driver class in zephyr, but the subsystem used is the `IPC` one.

jhedberg replied:
I change it to IPC, but I also left IPM as an alias, since there are several DT bindings referring to it which I don't feel comfortable touching this close to the release.

jhedberg committed:
Bluetooth: drivers: Rename IPM to IPC

This bus type was originally created for what's today the ipc.c HCI driver.
Since this type hasn't yet been synced with BlueZ, rename it for
consistency, however leave the old define to not break backwards
compatibility with existing DT bindings (there are several more that use
"ipm" than ipc.c).

Signed-off-by: Johan Hedberg <johan.hedberg@silabs.com>

alwa-nordic commented on dts/bindings/bluetooth/bt-hci.yaml:
```diff
@@ -20,6 +20,7 @@ properties:
       - "spi"
       - "i2c"
       - "ipm"
```
Would this be useful?

```suggestion
      - "ipm"  # Deprecated alias for "ipc"
```

jhedberg replied:
Done.

jhedberg commented:
@alwa-nordic @Thalley sorry, for the sake of completeness I realized "smd" and "virtio" need to be in the allowed DT property values as well, in case some future HCI driver/binding wants to use those. So you'll need to renew your approvals.

Vudentz commented:
> > Should we add a code comment with a link to the authoritative source?
> 
> I'm just not sure how to make that reference so that it stays up to date, besides simply saying "These values come from Linux/BlueZ". What makes the situation more confusing is that BlueZ has multiple places defining them. There's e.g. the following which seems to be out of date with the latest VirtIO addition (@Vudenz you might want to fix this): https://git.kernel.org/pub/scm/bluetooth/bluez.git/tree/src/shared/btsnoop.h#n51
> 
> The Linux kernel also has its own definitions, which could be seen as equally authoritative as the user space tree: https://git.kernel.org/pub/scm/bluetooth/bluetooth-next.git/tree/include/net/bluetooth/hci.h#n60

I will update those value and add a reference to this issue so we crosscheck, Im also planning to add PCI(12) bus after IPC.
